prime-router/src/main/kotlin/fhirengine/engine/FHIRConverter.kt

```diff
@@ -144,10 +139,52 @@ class FHIRConverter(
                     ).serialize()
                 )
             }
-        } catch (e: IllegalArgumentException) {
+        }
+    }
+
+    /**
+     * Converts an HL7 message into FHIR bundles
+     *
+     * [message] incoming HL7 message to be converted into FHIR
+     * [actionLogger] keeps track of any validation errors when reading messages
+     *
+     * @return one or more FHIR bundles
+     */
+    internal fun getBundlesFromHL7(
+        message: RawSubmission,
+        actionLogger: ActionLogger
+    ): List<Bundle> {
+        // create the hl7 reader
+        val hl7Reader = HL7Reader(actionLogger)
+        // get the hl7 from the blob store
+        val hl7messages = hl7Reader.getMessages(message.downloadContent())
+
+        val bundles = if (actionLogger.hasErrors()) {
+            val e = java.lang.IllegalArgumentException(actionLogger.errors.joinToString("\n") { it.detail.message })
```

Comment:
maybe change the order here.  Create a val with the message then pass that to the logger and actionLogger.  There is no real exception here besides the message that was created.  Then last throw the exception with the message.  We talked about removing this exception, but that can be another ticket (let's create the ticket).